test/emitter.test.js

```diff
@@ -18,6 +18,49 @@ describe('Emitter', function () {
     })
   })
 
+  describe('#currying', function () {
+    let remit, emitter
+
+    before(function () {
+      remit = Remit()
+      emitter = remit.emit('foo')
+    })
+
+    it('should curry when not exhausted', function (done) {
+      remit
+        .listen('had_jalfrezi')
+        .handler(async (event) => {
+          try {
+            expect(event.data).to.equal('had a jalfrezi')
+            done()
+          } catch (err) {
+            done(err)
+          }
+
+          return event.data
+        })
+        .start()
+        .then(_ => remit.emit('had_jalfrezi')('had a jalfrezi'))
```

Comment:
Need to check that what comes back here is a promise that resolves once the emission has been sent as opposed to just an Emission object.

Reply:
How would i be able to go about that? Not sure theres any way to tell if emission is received or not. Could log Date.now in the endpoint and after resolution, is that enough, is it flimsy?

Reply:
Ah no it’s literally just checking that `remit.emit` returns a _promise_ (for the emission being sent) or an _Emission_ object when appropriate, to check that the currying it doing the correct thing. :)

No need to worry about if the emission was received or not.

Reply:
Oh! Alright i'll take a look!